source/databricks/calculation-engine/tests/integration/calculation_input/test_calculation_input_reader.py

```diff
@@ -118,3 +170,127 @@ def test___read_metering_point_periods__returns_df_with_correct_settlemet_method
+    spark: SparkSession,
+    settlement_method: InputSettlementMethod,
+    expected: SettlementMethod,
+) -> None:
+    row = _create_metering_point_period_row(settlement_method=settlement_method)
+    df = spark.createDataFrame(data=[row], schema=metering_point_period_schema)
+    sut = CalculationInputReader(spark)
+
+    # Act
+    with mock.patch.object(sut, "_read_table", return_value=df):
+        actual = sut.read_metering_point_periods()
+
+    # Assert
+    assert actual.collect()[0][Colname.settlement_method] == expected.value
+
+
+@pytest.mark.parametrize(
+    "expected_schema, method_name, create_row",
+    [
+        (
+            metering_point_period_schema,
+            CalculationInputReader.read_metering_point_periods,
+            _create_metering_point_period_row,
+        ),
+        (
+            time_series_point_schema,
+            CalculationInputReader.read_time_series_points,
+            _create_time_series_point_row,
+        ),
+        (
+            charge_master_data_periods_schema,
+            CalculationInputReader.read_charge_master_data_periods,
+            _create_charge_master_period_row,
+        ),
+        (
+            charge_link_periods_schema,
+            CalculationInputReader.read_charge_links_periods,
+            _create_charge_link_period_row,
+        ),
+        (
+            charge_price_points_schema,
+            CalculationInputReader.read_charge_price_points,
+            _create_change_price_point_row,
+        ),
+    ],
+)
+def test__read_data__returns_df(
+    spark: SparkSession, expected_schema: StructType, method_name: str, create_row: any
+) -> None:
+    # Arrange
+    row = create_row()
+    sut = CalculationInputReader(spark)
+    df = spark.createDataFrame(data=[row], schema=expected_schema)
+    method = getattr(sut, str(method_name.__name__))
+
+    # Act & Assert
+    with mock.patch.object(sut, "_read_table", return_value=df):
+        method()
+
+
+@pytest.mark.parametrize(
+    "expected_schema, method_name, create_row",
+    [
+        (
+            metering_point_period_schema,
+            CalculationInputReader.read_metering_point_periods,
+            _create_metering_point_period_row,
+        ),
+        (
+            time_series_point_schema,
+            CalculationInputReader.read_time_series_points,
+            _create_time_series_point_row,
+        ),
+        (
+            charge_master_data_periods_schema,
+            CalculationInputReader.read_charge_master_data_periods,
+            _create_charge_master_period_row,
+        ),
+        (
+            charge_link_periods_schema,
+            CalculationInputReader.read_charge_links_periods,
+            _create_charge_link_period_row,
+        ),
+        (
+            charge_price_points_schema,
+            CalculationInputReader.read_charge_price_points,
+            _create_change_price_point_row,
+        ),
+    ],
+)
+def test__read_data__throws_exception_when_schema_mismatch(
```

Comment:
I believe that the term in python is "raise" (as opposed to "throw").
```suggestion
def test__read_data__raises_value_error_when_schema_mismatch(
```